recipes-ni/ni-hw-support/ni-hw-support-common_1.0.bb

```diff
@@ -0,0 +1,54 @@
+SUMMARY = "Support scripts and utilities common to all NI hardware products"
+DESCRIPTION = "Support scripts and utilities for all NI hardware products which are supported by NI LinuxRT."
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+LICENSE = "MIT"
+SECTION = "base"
+
+DEPENDS += "\
+	update-rc.d-native \
+"
+
+SRC_URI += "\
+	file://init.d/ni-rename-ifaces \
+"
+
+S = "${WORKDIR}"
+
+
+do_install () {
+	install -d ${D}${sysconfdir}/init.d/
+
+	install -m 0755 ${S}/init.d/ni-rename-ifaces ${D}${sysconfdir}/init.d/ni-rename-ifaces
+}
+
+pkg_postinst_${PN} () {
+	if [ -n "$D" ]; then
+		OPT="-r $D"
+	else
+		OPT="-s"
+	fi
+
+	update-rc.d $OPT ni-rename-ifaces start 38 S .
```

Comment:
Why is this not part of the install stage? Asking to learn the difference, because other recipes call update-rc.d in install.

Reply:
>Why is this not part of the install stage? 

Execing `update-rc` in the `do_install` means that the `data.tar.gz` in the IPK will contain the `/etc/rc*.d` symlink. When you install the IPK, okpg will drop the symlink in the correct place; so in the simplest case, it is functionally equivalent. But putting the rc*.d symlink in the data section isn't *correct* in general package management, since you normally can't be sure that the user hasn't reconfigured how their init system works. It's more portable to let their native `update-rc.d` tooling do its job.

>because other recipes like initscripts-nilrt call update-rc.d in install, and do not call update-rc.d remove (bug?).

I would say `initscripts-nilrt` is incorrect, for the same reasons above. But if the RC-symlink is in the data section of the IPKS, then opkg should remove it at uninstall-time. So, explicitly calling `update-rc` in the postrm is unnecessary (but still wrong!) in that case.